demos/rest/rest.ps1

```diff
@@ -1,46 +1,43 @@
-﻿# NOTE:  This demo is still in progress and needs validation in Linux
-# ------------------------------------
-
-#region Setup the credentials for use in HTTP header
-$user = '<insert GitHub PAT token>'
-$pass= ""
-$pair = "${user}:${pass}"
-$bytes =  [System.Text.Encoding]::ASCII.GetBytes($pair)
-$base64 = [System.Convert]::ToBase64String($bytes)
-$basicAuthValue = "Basic $base64"
-$headers = @{ Authorization = $basicAuthValue }
-#endregion
-
-# Changing the status of a Private GitHub repository to Public
-
-# URL to PowerShell Github Repo
-$PowerShellGithubUri = '<insert GitHub repo URL>'
-
-# Get the blob from the Github API as a PS object
-$JsonBlock = Invoke-RestMethod -Uri $PowerShellGithubUri -Headers $headers
-
-# Explore the object (Notice that it is a private repo)
-$JsonBlock
-
-# Given it is an object, you can explore and interact with it
-# Change the private value to false
-$JsonBlock.private = 'false'
-
-# Convert the object back to a json
-$Json = ConvertTo-Json $JsonBlock
-
-# Post the updated json block back to the GitHub
-Invoke-RestMethod -Uri $PowerShellGithubUri -Headers $headers -Method Post -Body $Json
-
-
-# --------------
-
-# We can also use the PS objects to sort the different repos on github
-
-# If we grab the json from the PowerShell github
-Invoke-RestMethod https://api.github.com/users/powershell/repos | sv repoData
-
-# We can sort it based on the number of forks each repo has
-$repoData | Sort-Object -Property forks_count -Descending | ft -f id,name,stargazers_count,forks_count
-
-$repoData | Sort-Object -Property stargazers_count -Descending | ft -f id,name,stargazers_count,forks_count
+﻿
+#-----------------
+
+function Get-Issues
+{
+    param([string]$UserName,
+          [string]$Repo,
+          [ValidateRange(1,100)][int]$PerPage = 100)
+
+    $body = @{
+        per_page = $PerPage
+    }
+
+    $uri = "https://api.github.com/repos/$UserName/$Repo/issues"
+    while ($uri)
+    {
+        $response = Invoke-WebRequest -Uri $uri -Body $body
```

Comment:
Maybe there's a cleaner way, I'm certainly no expert here, but I needed the Headers too to follow the next link, so that's how I ended up with Invoke-WebRequest.